algorithms/kernel/kernel.h

```diff
@@ -54,13 +54,13 @@
         ((KernelClass<templateArguments, cpu> *)(_kernel))->method(__VA_ARGS__);
 
 #define __DAAL_INSTANTIATE_DISPATCH_IMPL(ContainerTemplate, Mode, ClassName, BaseClassName, ...) \
-extern template class DAAL_KERNEL_SSE2_CONTAINER1(ContainerTemplate, __VA_ARGS__);               \
-extern template class DAAL_KERNEL_SSSE3_CONTAINER1(ContainerTemplate, __VA_ARGS__);              \
-extern template class DAAL_KERNEL_SSE42_CONTAINER1(ContainerTemplate, __VA_ARGS__);              \
-extern template class DAAL_KERNEL_AVX_CONTAINER1(ContainerTemplate, __VA_ARGS__);                \
-extern template class DAAL_KERNEL_AVX2_CONTAINER1(ContainerTemplate, __VA_ARGS__);               \
-extern template class DAAL_KERNEL_AVX512_MIC_CONTAINER1(ContainerTemplate, __VA_ARGS__);         \
-extern template class DAAL_KERNEL_AVX512_CONTAINER1(ContainerTemplate, __VA_ARGS__);             \
```

Comment:
What is the reason for the changes?

Reply:
Together with changes in include/services/daal_kernel_defines.h, it fixes a bug of building on one particular processor core. I can separate this change from my pull request.

Reply:
It was modified by Andrey Gilev to reduce the size of the library with avoiding unnecessary template instantiation.  Need to discuss this solution with Gilev Andrey.